integration/all_stores_test.go

```diff
@@ -0,0 +1,94 @@
+// +build integration
+
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/integration/e2e"
+	e2edb "github.com/cortexproject/cortex/integration/e2e/db"
+	"github.com/cortexproject/cortex/integration/e2ecortex"
+)
+
+func TestAllIndexStores(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	// Start dependencies.
+	dynamo := e2edb.NewDynamoDB()
+	bigtable := e2edb.NewBigtable()
+
+	stores := []string{"aws-dynamo", "bigtable"}
+	perStoreDuration := 14 * 24 * time.Hour
+
+	consul := e2edb.NewConsul()
+	require.NoError(t, s.StartAndWaitReady(dynamo, bigtable, consul))
+
+	// lets build config for each type of Index Store.
+	now := time.Now()
+	oldestStoreStartTime := now.Add(time.Duration(-len(stores)) * perStoreDuration)
+
+	storeConfigs := make([]storeConfig, len(stores))
+	for i, store := range stores {
+		storeConfigs[i] = storeConfig{From: oldestStoreStartTime.Add(time.Duration(i) * perStoreDuration).Format("2006-01-02"), IndexStore: store}
+	}
+
+	// bigtable client needs to set an environment variable when connecting to an emulator
+	bigtableFlag := map[string]string{"BIGTABLE_EMULATOR_HOST": bigtable.NetworkHTTPEndpoint()}
+
+	// here we are starting and stopping table manager for each index store
+	// this is a workaround to make table manager create tables for each config since it considers only latest schema config while creating tables
+	for i := range storeConfigs {
```

Comment:
This looks a bit hacky. Why don't we run the table manager only once, but with a schema config containing both storages (two entries in the same schema config)? Doing it with `buildSchemaConfigWith()` may be engineered (would require extra refactoring I'm not sure it's worth). You could just rollback `buildSchemaConfigWith()` and have a static schema config defined in this file.

Reply:
This is done because `TableManager` only uses latest config to create tables i.e it does not consider older configs while creating tables. There is a PR open to fix this behaviour in Cortex. See https://github.com/cortexproject/cortex/pull/1446

We have not seen someone complain about it because we add only one schema at a time to cortex as and when needed but here I am setting multiple schema configs at a time.
Does it make sense?

Reply:
Yes. I wasn't aware of the issue. Could you add a `// TODO` comment referencing the PR and saying that we can merge two two schema configs into 1 and just run the table-manager once once that PR is merged, please?